src/AuthenticationResponse.ts

```diff
@@ -39,6 +41,16 @@ export default class AuthenticationResponse {
     verifiedJwt: SIOP.VerifiedAuthenticationRequestWithJWT,
     responseOpts: SIOP.AuthenticationResponseOpts
   ): Promise<SIOP.AuthenticationResponseWithJWT> {
+    const pejs: PEJS = new PEJS();
+    // bunch of ifs to make sure we need to call evaluate
+    const pd: PresentationDefinition =
+      verifiedJwt.payload.claims['id_token']['verifiable_presentations'].presentation_definition;
+    const result: EvaluationResults = pejs.evaluate(pd, responseOpts.vp);
+    if (result.errors.length) {
+      throw new Error(`${SIOPErrors.EVALUATE_PRSENTATION_EXCHANGE_FAILED}`);
+    }
+    const ps = pejs.submissionFrom(pd, responseOpts.vp.getVerifiableCredentials());
```

Comment:
We can start out this way for now and because of the deadline, but we really need to have a split here.

Basically we need to expose the submission requirements/definition. Then have the wallet/holder supply the the credentials to the pe-js filtering mech. Then the user selects and/or approves the actual credentials. Only after that we can generate the AuthResponse.